core/app/models/spree/taxon.rb

```diff
@@ -85,6 +85,16 @@ def active_products
       products.not_deleted.available
     end
 
+    # @return [ActiveRecord::Relation<Spree::Product>] all self and descendant products
+    def all_products
+      Product.joins(:taxons).where(spree_taxons: { id: descendant_ids.push(id) })
```

Comment:
It would be better to use `descendant_ids + [id]`.

Push would append to the array `descendant_ids` returns which we want to avoid (I don't know whether or not that would cause a real issue, but I can't know without reading its source, so we want to avoid it)